flytepropeller/pkg/controller/nodes/task/k8s/plugin_manager_test.go

```diff
@@ -1015,15 +1036,16 @@ func TestFinalize(t *testing.T) {
 		tctx := getMockTaskContext(PluginPhaseStarted, PluginPhaseStarted)
 		o := &v1.Pod{
 			ObjectMeta: metav1.ObjectMeta{
-				Name:      tctx.TaskExecutionMetadata().GetTaskExecutionID().GetGeneratedName(),
-				Namespace: tctx.TaskExecutionMetadata().GetNamespace(),
+				Name:       tctx.TaskExecutionMetadata().GetTaskExecutionID().GetGeneratedName(),
+				Namespace:  tctx.TaskExecutionMetadata().GetNamespace(),
+				Finalizers: []string{finalizer},
 			},
 		}
 
 		assert.NoError(t, fakeKubeClient.GetClient().Create(ctx, o))
 
 		p.OnBuildIdentityResource(ctx, tctx.TaskExecutionMetadata()).Return(o, nil)
-		pluginManager := PluginManager{plugin: &p, kubeClient: fakeKubeClient}
+		pluginManager := PluginManager{plugin: &p, kubeClient: fakeKubeClient, updateBackoffRetries: 5}
```

Comment:
This `updateBackoffRetries` parameter ends up being `e.updateBackoffRetries` in
```
	retryBackoff := wait.Backoff{
		Duration: time.Duration(e.updateBaseBackoffDuration) * time.Millisecond,
		Factor:   2.0,
		Jitter:   0.1,
		Steps:    e.updateBackoffRetries,
	}
```
If this is unset (which it was in tests), then `Steps: 0` which means `_ = wait.ExponentialBackoff(retryBackoff, func() (bool, error) {` inside `Finalize()` times out almost immediately, which effectively means `e.clearFinalizer()` is never called and the finalizer is never removed. The default setting is 5 but I think it's too dangerous a) to allow users to tweak these backoff settings, or b) to use an exponential backoff at all.

Thoughts?

Reply:
The [original](https://github.com/flyteorg/flyte/pull/5673) context of this change had to do with informers getting stale info in the case of array nodes. 

cc: @pvditt 

> to allow users to tweak these backoff settings

Do you mean we should validate that it's a strictly positive value, right?

> to use an exponential backoff at all.

Can you expand on that?